File: Lazy Initialization/Python/Lazy.py
class Lazy:
  def __init__(self, *args):
    self._initialized = false;
    self._data = args;
  
  def _callInit(self):
    if not self._initialized:
      self._initialized = true;
      _lazyInit(*self._data); #(check how to unzip?)
  
  # Inserire qui eventuali variabili statiche
  
  def _lazyInit(self, *args):
    # Inserire qui l'inizializzazione
    pass;
  
  def wakeup(self):
    _callInit();
    # Inserire logica metodo

File: Lazy Initialization/Java/Lazy.java
public class Lazy {
  
  //// NON MODIFICARE ////
  private boolean initialized = false;
  private Object[] data; // dati per l'inizializzazione dell'istanza
  
  // Costruttore, tiene in memoria i dati per l'inizializzazione
  public void Lazy(Object... args) {
    this.data = args;
  }
  
  private void callInit() {
    if !(this.initialized) {
      this.initialized = true;
      lazyInit(this.data);
    }
  }
  
  //// MODIFICARE DA QUI IN POI ////
  
  // Inserire qui eventuali variabili
  
  private void lazyInit(Object... args) {
    // Gestire qui inizializzazione
  }
  
  public void wakeup() {
    callInit();
    // Inserire logica funzione
  }
}
